.github/workflows/post_release.yaml

```diff
@@ -0,0 +1,65 @@
+name: Post Release Automation
+
+on:
+  release:
+    types: [published]
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-version:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Configure git user for actions
+        run: |
+          git config --global user.email "action@github.com"
+          git config --global user.name "GitHub Action"
+
+      - name: Set environment variables
+        run: |
+          echo "NEW_VERSION=${{ github.event.release.tag_name }}" >> $GITHUB_ENV
```

Comment:
I think the problem is probably going to be that the tag points to a commit that is parent to the commit that the created PR is trying to merge, or I might be misunderstanding something.

Maybe if we do instead a manually triggered workflow that takes the new version number as parameter and does everything, even creating the tag, automatically?

Reply:
I thought on this a bit. This is true, these changes would create a PR to main after the release. So they would not be included in the release - I am not sure what was going through my mind here. Ideally we do this before the release, so everything is included before we publish.

Following your comment:
> Maybe if we do instead a manually triggered workflow that takes the new version number as parameter and does everything, even creating the tag, automatically?

I am thinking we first rename/move `CHANGELOG.md` to `RELEASE_NOTES.md`, this would now hold all the release notes and changes for every release so far. We then use`CHANGELOG.md` as the current log and release notes for what is being worked on. So in a workflow we can simply copy the contents of the changelog to the full set of release notes.

The release process could then be:

1) We update the new changelog file with everything we want to include in the new release notes, including the description. This would be the final manual PR before drafting a new release, where it allows us all a way to review what the release will look like. This PR would be labelled with something like `release`, such that it can be used to trigger a workflow when merged.

2) When this PR is merged it would start the `pre_release.yaml` workflow.  This would essentially automate all the relavant changes that have to occur before a release. First the changelog would be copied into the cumulative release notes file, and the changelog template copied into the changelog file. Secondly the version would be bumped within `setup.cfg`. This would then create a PR to main (as we are already doing here) allowing us to review these changes quickly. 

3) When the automated`pre_release.yaml` PR is merged to main, this would then trigger the `draft_release.yaml` workflow. Essentially doing what you suggested. Extract the new version, create/push the new tag with the version, and draft the new release copying the contents within the release notes. We can then review/edit the draft and publish.

What do you think? It should be pretty easy for me to add now :)

Reply:
I don't know how easy it would be to detect the PR name, with manually trigger I meant to go to https://github.com/ethereum/execution-spec-tests/actions, and under "Workflows", like in this image:
![image](https://github.com/ethereum/execution-spec-tests/assets/11726710/eb692f04-bfb9-4ab9-9044-245fd8353c84)

We can create a new workflow that is called "Prepare Release and Tag", which takes the version number here:

![image](https://github.com/ethereum/execution-spec-tests/assets/11726710/47cc8762-c4bb-49e3-a585-f5cd98f750a8)

Does the automatic commit with all the appropriate changes, and pushes to main (which could be dangerous but we can bail on any minimal error) and then creates the tag that points to the commit that was just pushed.